packages/teams-js/test/public/pages.spec.ts

```diff
@@ -24,98 +29,216 @@ describe('AppSDK-TeamsAPIs', () => {
+    it('pages.setCurrentFrame should not allow calls before initialization', () => {
+      expect(() => pages.setCurrentFrame(frameContext)).toThrowError('The library has not yet been initialized');
+    });
 
-      const executeDeepLinkMessage = utils.findMessageByFunc('executeDeepLink');
-      utils.respondToMessage(executeDeepLinkMessage, true);
-      await promise;
+    Object.values(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+        it(`pages.setCurrentFrame should successfully set frame context when initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          pages.setCurrentFrame(frameContext);
+          let message = utils.findMessageByFunc('setFrameContext');
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toBe(frameContext);
+        });
+      } else {
+        it(`pages.setCurrentFrame should not allow calls from ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => pages.setCurrentFrame(frameContext)).toThrowError(
+            `This call is only allowed in following contexts: ${JSON.stringify(
+              allowedContexts,
+            )}. Current context: "${context}".`,
+          );
+        });
+      }
+    });
+  });
 
-      expect(executeDeepLinkMessage).not.toBeNull();
-      expect(executeDeepLinkMessage.args[0]).toBe(
-        'https://teams.microsoft.com/l/entity/fe4a8eba-2a31-4737-8e33-e5fae6fee194/tasklist123?webUrl=https%3A%2F%2Ftasklist.example.com%2F123&context=%7B%22channelId%22%3A%2219%3Acbe3683f25094106b826c9cada3afbe0%40thread.skype%22%2C%22subEntityId%22%3A%22task456%22%7D',
-      );
+  describe('Testing pages.initializeWithFrameContext function', () => {
+    const allowedContexts = [FrameContexts.content];
+    let frameContext: FrameInfo = {
+      contentUrl: 'someContentUrl',
+      websiteUrl: 'someWebsiteUrl',
+    };
+
+    Object.values(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+        it('pages.initializeWithFrameContext should successfully initialize and set the frame context', async () => {
+          await utils.initializeWithContext(context);
+          pages.initializeWithFrameContext(frameContext);
+          expect(utils.processMessage).toBeDefined();
+          expect(utils.messages.length).toBe(2);
+
+          let initMessage = utils.findMessageByFunc('initialize');
+          expect(initMessage).not.toBeNull();
+          expect(initMessage.id).toBe(0);
+          expect(initMessage.func).toBe('initialize');
+          expect(initMessage.args.length).toEqual(1);
+          expect(initMessage.args[0]).toEqual(version);
+          let message = utils.findMessageByFunc('setFrameContext');
```

Comment:
are initMessage or message getting reassigned and i'm missing it?

Reply:
We are just checking that the messages are initialized correctly and have expected arguments.